rmw_connext_cpp/src/functions.cpp

```diff
@@ -649,36 +678,21 @@ rmw_create_subscription(const rmw_node_t * node,
     goto fail;
   }
 
-  // allocates memory, but doesn't have to be freed.
-  // partition operater takes ownership of it.
-  name_tokens = rcutils_split_last(topic_name, '/');
-  if (name_tokens.size == 1) {
-    partition_str = DDS_String_dup(ros_topics_prefix);
-    topic_str = DDS_String_dup(name_tokens.data[0]);
-  } else if (name_tokens.size == 2) {
-    // TODO(Karsten1987): Fix utility function for this
-    size_t partition_length = strlen(ros_topics_prefix) + strlen(name_tokens.data[0]) + 2;
-    char * concat_str = reinterpret_cast<char *>(rmw_allocate(partition_length * sizeof(char)));
-    snprintf(concat_str, partition_length, "%s/%s", ros_topics_prefix, name_tokens.data[0]);
-    // Connext will call deallocate on this, passing ownership to connext
-    partition_str = DDS_String_dup(concat_str);
-    topic_str = DDS_String_dup(name_tokens.data[1]);
-    // free temporary memory
-    rmw_free(concat_str);
-  } else {
-    RMW_SET_ERROR_MSG("Illformated topic name");
+  if (!_process_topic_name(
+      topic_name,
+      qos_profile->avoid_ros_namespace_conventions,
+      &topic_str,
```

Comment:
@wjwwood When is the allocated memory of this variable being freed? Same for `partition_str` next line.